File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.py
class Solution:
    def numSpecial(self, mat: List[List[int]]) -> int:
        rows = [0] * len(mat)
        cols = [0] * len(mat[0])
        res = 0
        for r in range(0, len(mat)):
            for c in range(0, len(mat[r])):
                if mat[r][c] == 1:
                    rows[r] += 1
                    cols[c] += 1
        for r in range(0, len(mat)):
            for c in range(0, len(mat[r])):
                if mat[r][c] == 1 and rows[r] == 1 and cols[c] == 1:
                    res += 1
        return res

File: 1582-special-positions-in-a-binary-matrix/1582-special-positions-in-a-binary-matrix.kt
class Solution {
    fun numSpecial(mat: Array<IntArray>): Int {
        val rows = IntArray(mat.size) {0}
        val cols = IntArray(mat[0].size) {0}
        var res = 0
        for (r in 0..(mat.size-1)) {
            for (c in 0..(mat[r].size-1)) {
                if (mat[r][c] === 1) {
                    rows[r]++
                    cols[c]++
                }
            }
        }
        for (r in 0..(mat.size-1)) {
            for (c in 0..(mat[r].size-1)) {
                if (mat[r][c] === 1 && rows[r] === 1 && cols[c] === 1) {
                    res++
                }
            }
        }
        return res
    }
}